pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -838,6 +838,29 @@ func writeUnreplicatedSST(
 	)
 	defer unreplicatedSST.Close()
 
+	clearedSpan, err := rewriteRaftState(ctx, id, hs, kvserverpb.RaftTruncatedState{
```

Comment:
While here, could add more comments:

- why picking the `RaftTruncatedState` equal to the snapshot entry ID
- this func also clears the raft log (which is fine when we're applying a snapshot, as long as the state machine is synced first or with this write).

Reply:
Done.